### PR TITLE
updateSetting e2e typed helper

### PR DESCRIPTION
**Pull request by npretto**
Note: The first commit has the actual function, the second commit changes all the E2Es to use it.

# Description 
I wasted at least half an hour today debugging a e2e test, not realizing the the BE was returning 204 even though i was not sending the correct values to the endpoint.
(I was updating `application-font-files` and sending the array of fonts directly in the body instead of wrapping it in an object like `{value: []}`.

This PR adds a putSetting helper which is typed and should not only make it impossible to do such mistake, but it also have autocompletion so it should make us bit more happy when working with E2Es.

image

Next possible step would be to make a `putSettings` that accepts multiple settings and just calls PUT `/api/setting`

### CODE REVIEW

File: e2e/support/helpers/api/putSetting.ts
Comment (oisincoveney) [inline e2e/support/helpers/api/putSetting.ts:1]: nittiest of nits: for files I think we prefer kebabs so `put-setting.ts`

Comment (npretto) [reply e2e/support/helpers/api/putSetting.ts:1]: image
we're not very consistent in the helper folder 😆, we use camelCase in the api folder but not outside

Comment (kamilmielnik) [reply e2e/support/helpers/api/putSetting.ts:1]: I introduced the `/api` folder and the logic I used is this: if a file exports only 1 function - it is called exactly like the function (with `camelCase`) - it's just faster to copy/paste the name as is.

I left the files that export multiple functions untouched.
BTW the `e2e-` prefix is redundant as they all live in `e2e` folder anyway.

**Review by kamilmielnik — COMMENTED** [review]
Awesome :+1:

File: e2e/support/helpers/api/putSetting.ts
Comment (npretto) [reply e2e/support/helpers/api/putSetting.ts:1]: @nemanjaglumac I don't have a strong opinion on it, but I wanted this to be different from the names we use in the core app, where I think use use "update"

**Review by kamilmielnik — APPROVED** [review]
Looks good :+1: 

FWIW I support [Nemanja's comment](https://github.com/metabase/metabase/pull/48161/files#r1778993591).

**Comment by npretto** [comment]
> Awesome 👍

`updateSetting` it is then :)

**Pull request merged**